### DISCUSSION TITLE
coupled with synxflow

### DISCUSSION THREAD

**Discussion by kinghuihui9999** [SynxFlow/SynxFlow#40] [Ideas]

**Comment by xiaxilin** [comment]
Hi, it is good that hear you plan to do this. It seems you are doing a one-way coupling?

I think adding overflow as rainfall is feasible. Do you also consider drainage loss within SynxFlow?

**Comment by kinghuihui9999** [comment]


**Reply by xiaxilin** [reply]
I have also moved this thread to discussions which is more relevant.

**Comment by xiaxilin** [comment]
To consider the drainage loss, a way is to define the 'sewer_sink' parameter, which takes negative values (m/s) to consider drainage loss. A more closely coupled approach would be to use SWMM's C API within the flood simulation engine by change the CUDA code.

_Additionally, I have observed that the areas of inundation are mainly concentrated around the nodes, which suggests that flooding due to terrain features might not be adequately represented._

I don't understand this, SynxFlow have already considered the effects of topography during the runoff routing. More details about the algorithm and applications can found in [this paper](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1002/2016WR020055)

**Comment by BlackDogo** [comment]
Hello @kinghuihui9999, have you sucessfully coupled SWMM with SynxFlow? I'm interested to know more about this coupling work. Thanks

**Comment by kinghuihui9999** [comment]
yes
